Bug: Crash in `Repo._load_git_data` when `.git-blame-ignore-revs` does not exist

For older version of `git`, such as 2.34.1, `git blame` errors out if `.git-blame-ignore-revs` does not exist. A crash occurs in [repo.py line 271](https://github.com/TheRenegadeCoder/subete/blob/main/subete/repo.py#L271) due to this issue. I guess the reason why this hasn't been an issue is that GitHub seems to be using `git 2.39.1`.

Here are the details of the crash when running the tests:
```
tests/test_integration.py:3: in <module>
    TEST_REPO = subete.load()
subete/__init__.py:21: in load
    return Repo(source_dir=source_dir)
subete/repo.py:35: in __init__
    self._load_git_data()
subete/repo.py:271: in _load_git_data
    blame = self._git_repo.blame('HEAD', f"{program._path}/{program._file_name}")
venv/lib/python3.7/site-packages/git/repo/base.py:876: in blame
    data = self.git.blame(rev, '--', file, p=True, stdout_as_string=False, **kwargs)
venv/lib/python3.7/site-packages/git/cmd.py:585: in <lambda>
    return lambda *args, **kwargs: self._call_process(name, *args, **kwargs)
venv/lib/python3.7/site-packages/git/cmd.py:1124: in _call_process
    return self.execute(call, **exec_kwargs)
venv/lib/python3.7/site-packages/git/cmd.py:928: in execute
    raise GitCommandError(redacted_command, status, stderr_value, stdout_value)
E   git.exc.GitCommandError: Cmd('git') failed due to: exit code(128)
E     cmdline: git blame -p HEAD -- /tmp/tmp6ouuhm62/archive/a/abap/hello-world.abap
E     stderr: 'fatal: could not open object name list: .git-blame-ignore-revs'
```

--- a/setup.py
+++ b/setup.py
@@ -13,7 +13,7 @@ with open("README.md", "r") as fh:
 
 MAJOR = 0
 MINOR = 12
-PATCH = 0
+PATCH = 1
 
 name = "subete"
 version = f"{MAJOR}.{MINOR}"

--- a/subete/repo.py
+++ b/subete/repo.py
@@ -264,6 +264,13 @@ class Repo:
         and inject that data into the repo object. This was done for simplicity.
         It seems like way more of a pain to try to pass the git data around.
         """
+
+        # Make sure .git-blame-ignore-revs exists for older versions of git and
+        # keep track of whether it existed before
+        blame_path = Path(f"{self._temp_dir.name}/.git-blame-ignore-revs")
+        blame_path_exists = blame_path.exists()
+        blame_path.touch()
+
         for language in self:
             language: LanguageCollection
             for program in language:
@@ -278,6 +285,9 @@ class Repo:
                 program._modified = max(times)
                 logger.info(f"Loaded git data into existing program ({program}): {program._created} - {program._modified} by {program._authors}")
 
+        # Delete .git-blame-ignore-revs if it did not exist before
+        if not blame_path_exists:
+            blame_path.unlink()
 
 class LanguageCollection:
     """

